app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyPresenter.java

```diff
@@ -194,6 +191,34 @@ private Completable showPendingTransaction(Payment transaction) {
     }
   }
 
+  private void handleFraudFlow() {
+    disposables.add(onChainBuyInteract.isWalletBlocked()
+        .subscribeOn(networkScheduler)
+        .flatMap(blocked -> {
```

Comment:
U forgot to use observeOn(networkScheduler) here

Reply:
isn't it already in network due to subscribeOn?